### PR TITLE
End game

### PR DESCRIPTION
**Pull request by OsvaldoPrajitno**
# An extension of the getlocation branch.
Instead of creating and saving a *Game* object in the startGame endpoint, the *Game* object is created **after** the game is finished. Chose to do it this way bcs all the game logic is happening in the frontend.

### CODE REVIEW

File: backend/src/index.ts [hunk 1]
```diff
@@ -61,6 +69,16 @@ const session_remove = async (sessionId: string) => {
   return true;
 };
 
+const sessionIdToUserId = async (sessionId: string) => {
+  const sessionData = query(sessions, where("sessionId", "==", sessionId));
+  const session = await getDocs(sessionData);
+
+  if (session.empty) {
+    return "guest";
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: i think i'd prefer if this either:
a) returns `undefined`
b) returns a constant named `guest` that equals `undefined`

File: backend/src/index.ts [hunk 2]
```diff
@@ -191,6 +209,55 @@ app.post("/login", async (req: TypedRequest<LoginBody>, res: Response) => {
...
+  "endGame", 
+  async (
+  req: TypedRequest<{gameMode: Gamemode, levels: Level["id"][], score: number}>, 
+  res: Response
+  ) => {
+    const { gameMode, levels, score } = req.body;
+  
+    const userId = await sessionIdToUserId(req.sessionID);
+
+      if (userId !== "guest") {
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: i don't think this should be indented - run thru prettier

Comment (lachlanshoesmith) [reply backend/src/index.ts]: i'd also recommend changing to something like
`if (userId === guest) ...` and return early

**Review by lachlanshoesmith — CHANGES_REQUESTED** [review]
i think yellowbutton can be removed unless it's being used elsewhere
seems like there's some weird stuff going on with different branches combining that include it lol

**Comment by OsvaldoPrajitno** [comment]
> i think yellowbutton can be removed unless it's being used elsewhere seems like there's some weird stuff going on with different branches combining that include it lol

@lachlanshoesmith do you mean remove the entire component folder?

**Comment by lachlanshoesmith** [comment]
> > i think yellowbutton can be removed unless it's being used elsewhere seems like there's some weird stuff going on with different branches combining that include it lol
> 
> @lachlanshoesmith do you mean remove the entire component folder?

so long as the component isn't being used, yes

**Pull request merged**